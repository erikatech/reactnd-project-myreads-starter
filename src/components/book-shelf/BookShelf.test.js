import React from 'react';
import { shallow, mount } from 'enzyme';
import BookShelf from './BookShelf';

let shelf = null;

beforeAll(() => {
	shelf = {title: 'Currently Reading', shelf: "currentlyReading", books: [{id: 1, shelf: 'currentlyReading'}]};
});

describe('BookShelf testing', () => {

	it('should render without crashing', () => {
		expect(shallow(
		  <BookShelf
		    shelf={shelf}/>
		)).toMatchSnapshot();
	});


	it('should render the right amount of shelves', () => {
		const moveBook = jest.fn();
		const bookShelf = mount(
		  <BookShelf
			  shelf={shelf}
			  key={shelf.title}
			  moveBook={moveBook}
			/>
		);
		expect(bookShelf.find('li').length).toBe(1);
	});

	it('should call the method when changes the select', () => {
		const moveBook = jest.fn();
		const bookShelf = mount(
		  <BookShelf
			shelf={shelf}
			key={shelf.title}
			moveBook={moveBook}
		  />
		);
		bookShelf.find('select').simulate('change', {target: {value : 'read'}});
		expect(moveBook).toHaveBeenCalledTimes(1);
	});

	it('should have four options to classify books', () => {
		const bookShelf = mount(<BookShelf
		  shelf={shelf}
		  key={shelf.title}/>);
		expect(bookShelf.find('option').length).toBe(5);
	});

	it('should not have authors rendered', () => {
		const bookShelf = mount(<BookShelf
		  shelf={shelf}
		  key={shelf.title}/>);
		expect(bookShelf.find('book-authors').length).toBe(0);
	});
});