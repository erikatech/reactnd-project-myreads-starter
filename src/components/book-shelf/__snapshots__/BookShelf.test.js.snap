// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookShelf testing should render without crashing 1`] = `
<div
  className="list-books-content"
>
  <div
    className="bookshelf"
  >
    <h2
      className="bookshelf-title"
    >
      Currently Reading
    </h2>
    <div
      className="bookshelf-books"
    >
      <ol
        className="books-grid"
      >
        <li
          key="1"
        >
          <div
            className="book"
          >
            <div
              className="book-top"
            >
              <div
                className="book-cover"
                style={
                  Object {
                    "backgroundImage": "url(undefined)",
                    "height": 193,
                    "width": 128,
                  }
                }
              />
              <div
                className="book-shelf-changer"
              >
                <select
                  onChange={[Function]}
                  value="currentlyReading"
                >
                  <option
                    className=""
                    disabled={true}
                    key="Move to..."
                    value="moveTo"
                  >
                    Move to...
                  </option>
                  <option
                    className="selected-book"
                    disabled={false}
                    key="Currently Reading"
                    value="currentlyReading"
                  >
                    Currently Reading
                  </option>
                  <option
                    className=""
                    disabled={false}
                    key="Want to Read"
                    value="wantToRead"
                  >
                    Want to Read
                  </option>
                  <option
                    className=""
                    disabled={false}
                    key="Read"
                    value="read"
                  >
                    Read
                  </option>
                  <option
                    className=""
                    disabled={false}
                    key="None"
                    value="none"
                  >
                    None
                  </option>
                </select>
              </div>
            </div>
            <div
              className="book-title"
            />
          </div>
        </li>
      </ol>
    </div>
  </div>
</div>
`;
